kolibri/core/apps.py

```diff
@@ -0,0 +1,16 @@
+from __future__ import unicode_literals
+
+from django.apps import AppConfig
+
+
+class KolibriCoreConfig(AppConfig):
+    name = 'kolibri.core'
+
+    def ready(self):
+        from django.db import connection
+
+        # Only run the PRAGMAs if we're using SQLite
+        if connection.vendor == "sqlite":
+            cursor = connection.cursor()
+            cursor.execute("PRAGMA synchronous=OFF;")
```

Comment:
Reading the SQLite documentation on the Write Ahead Log, it does sound like using it would be a good solution if the issues we are facing are due to reads being blocked due to synchronous writes blocking access. https://www.sqlite.org/wal.html

Using Write Ahead Log, along with Synchronous set to NORMAL, should allow writes to happen to the WAL, while reads can still happen from the main database - the writes to the WAL are then synchronized to the database (by default) when the WAL reaches a size of 1000 pages. This is done in 'PASSIVE' mode by default, which means that if the database connection is busy then it won't interfere. This should have the nice effect of allowing cleanup to happen when the database is not too busy.

Thoughts?